xcrypto/rsa.go

```diff
@@ -101,18 +102,35 @@ func (k *PrivateKeyRSA) withKey(f func(C.SecKeyRef) C.int) C.int {
 
 // DecryptRSAOAEP decrypts data using RSA-OAEP.
 func DecryptRSAOAEP(h hash.Hash, priv *PrivateKeyRSA, ciphertext, label []byte) ([]byte, error) {
+	plaintext, err := evpDecrypt(priv.withKey, algorithmTypeOAEP, ciphertext, h)
+	if err != nil {
+		return nil, err
+	}
+	// If a label was provided, validate it
 	if len(label) > 0 {
-		// https://github.com/microsoft/go-crypto-darwin/issues/22
-		panic("crypto/rsa: label is not supported on macOS")
+		h.Write(label)
+		labelHash := h.Sum(nil)
+		h.Reset()
+		if len(plaintext) < len(labelHash) {
+			return nil, errors.New("invalid ciphertext: missing label hash")
+		}
+		// Extract and verify the label hash
+		if !bytes.Equal(plaintext[:len(labelHash)], labelHash) {
+			return nil, errors.New("invalid label hash")
+		}
```

Comment:
(Almost) never use `bytes.Equal` in crypto code, as it is vulnerable to oracle side channel attacks. Use [suble.ConstantTimeCompare](https://pkg.go.dev/crypto/subtle#ConstantTimeCompare). Also, don't short-circuit if the length are not equal, for the same reason.